tests/unit/test_inet.cpp

```diff
@@ -19,51 +19,266 @@
+  EXPECT_FALSE(lwpa_ip_is_multicast(&test_addr));
+
+  LWPA_IP_SET_V4_ADDRESS(&test_addr, 0xffffffff);
+  EXPECT_FALSE(lwpa_ip_is_multicast(&test_addr));
+
+  // Test the edges of the IPv6 multicast range
+  std::array<uint8_t, LWPA_IPV6_BYTES> v6_data = {0xfe, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+                                                  0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff};
+  LWPA_IP_SET_V6_ADDRESS(&test_addr, v6_data.data());
+  EXPECT_FALSE(lwpa_ip_is_multicast(&test_addr));
+
+  v6_data = {0xff, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+  LWPA_IP_SET_V6_ADDRESS(&test_addr, v6_data.data());
+  EXPECT_TRUE(lwpa_ip_is_multicast(&test_addr));
+
+  v6_data = {0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff};
+  LWPA_IP_SET_V6_ADDRESS(&test_addr, v6_data.data());
+  EXPECT_TRUE(lwpa_ip_is_multicast(&test_addr));
+}
+
+// Just tests reflexivity of the wildcard functions, we don't rely on the wildcard address to be
+// any specific value.
+TEST_F(InetTest, wildcard)
+{
+  LwpaIpAddr test_addr;
+
+  lwpa_ip_set_wildcard(kLwpaIpTypeV4, &test_addr);
+  EXPECT_TRUE(lwpa_ip_is_wildcard(&test_addr));
 
-  lwpaip_set_v4_address(&v4, 0x01020304);
-  ASSERT_TRUE(lwpaip_is_v4(&v4));
-  ASSERT_EQ(lwpaip_v4_address(&v4), 0x01020304u);
+  lwpa_ip_set_wildcard(kLwpaIpTypeV6, &test_addr);
+  EXPECT_TRUE(lwpa_ip_is_wildcard(&test_addr));
+}
+
+// Test the lwpa_ip_equal() and lwpa_ip_cmp() functions
+TEST_F(InetTest, compare)
+{
+  LwpaIpAddr v4;
+  LWPA_IP_SET_V4_ADDRESS(&v4, 0x01020304);
 
   // Copied address should compare equal
   LwpaIpAddr v4_copy = v4;
-  ASSERT_TRUE(lwpaip_equal(&v4_copy, &v4));
-  ASSERT_EQ(0, lwpaip_cmp(&v4_copy, &v4));
+  EXPECT_TRUE(lwpa_ip_equal(&v4_copy, &v4));
+  EXPECT_EQ(0, lwpa_ip_cmp(&v4_copy, &v4));
 
   LwpaIpAddr v6;
-
   std::array<uint8_t, LWPA_IPV6_BYTES> v6_data = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15};
-  lwpaip_set_v6_address(&v6, v6_data.data());
-  ASSERT_TRUE(lwpaip_is_v6(&v6));
-  ASSERT_EQ(0, memcmp(lwpaip_v6_address(&v6), v6_data.data(), LWPA_IPV6_BYTES));
+  LWPA_IP_SET_V6_ADDRESS(&v6, v6_data.data());
 
   // Copied address should compare equal
   LwpaIpAddr v6_copy = v6;
-  ASSERT_TRUE(lwpaip_equal(&v6_copy, &v6));
-  ASSERT_EQ(0, lwpaip_cmp(&v6_copy, &v6));
+  EXPECT_TRUE(lwpa_ip_equal(&v6_copy, &v6));
+  EXPECT_EQ(0, lwpa_ip_cmp(&v6_copy, &v6));
 
   // The v4 and v6 should not be equal, and per the comparison algorithm the v4 should be less than
   // the v6
-  ASSERT_FALSE(lwpaip_equal(&v4, &v6));
-  ASSERT_LT(lwpaip_cmp(&v4, &v6), 0);
+  EXPECT_FALSE(lwpa_ip_equal(&v4, &v6));
+  EXPECT_LT(lwpa_ip_cmp(&v4, &v6), 0);
 
-  // Further test the lwpaip_cmp() macro, with like address types
+  // Further test the lwpa_ip_cmp() function, with like address types
   LwpaIpAddr v4_less;
-  lwpaip_set_v4_address(&v4_less, 0x01020303);
+  LWPA_IP_SET_V4_ADDRESS(&v4_less, 0x01020303);
 
```

Comment:
You're not testing the case I mentioned earlier (the first octet has the high bit set)

Reply:
Will add